packages/oui-dropdown/src/dropdown-trigger.directive.js

```diff
@@ -0,0 +1,52 @@
+const dropdownTriggerClass = 'oui-dropdown__trigger'
+
+const defaultTrigger = `
+  <button type="text"
+    class="oui-button oui-button_dropdown oui-dropdown__trigger">
+    <span data-ng-bind="::$ctrl.text"></span>
+    <i class="oui-icon oui-icon-chevron-down" aria-hidden="true"></i>
+  </button>`
+
+export default ($compile, $log) => {
+  'ngInject'
+
+  return {
+    restrict: 'AE',
+    require: '^ouiDropdown',
+    controller: class {},
+    controllerAs: '$ctrl',
+    bindToController: {
+      text: '@?'
+    },
+    scope: {},
+    link: (scope, element, attrs, ctrl) => {
+      element.addClass(dropdownTriggerClass)
+
+      if (element[0].tagName.toLowerCase() === 'oui-dropdown-trigger') {
+        if (!attrs.text) {
+          $log.warn('You must define a text for a dropdown trigger.')
+        }
+
+        const compiled = $compile(defaultTrigger)
+        const triggerElement = compiled(scope)
+        element.replaceWith(triggerElement)
+        element = triggerElement
+      }
+
+      element.on('click', () => ctrl.onTriggerClick())
+      element.on('keydown', evt => ctrl.triggerKeyHandler(evt))
+      element.on('blur', evt => ctrl.triggerBlurHandler(evt))
+
+      element.attr({ 'aria-haspopup': true, 'aria-expanded': false })
+      scope.$watch(() => ctrl.isOpen(), open => {
```

Comment:
Manage aria-expanded is a good idea <3

Can you use a `click event` do manage it instead of `$scope.$watch` ?

Reply:
I don't know how to make it better here.